web/modules/custom/custom_weather/src/Plugin/Block/CustomWeather.php

```diff
@@ -33,7 +33,7 @@ class CustomWeather extends BlockBase implements ContainerFactoryPluginInterface
   /**
    * {@inheritdoc}
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, protected ConfigFactoryInterface $configFactory) {
+  public function __construct(protected ClientFactory $httpClient, array $configuration, $plugin_id, $plugin_definition, protected ConfigFactoryInterface $configFactory) {
```

Comment:
it is very weird to have your parameter in construct first, then all parameters as in parent, and then again your parameters
Please account for that in future.

Reply:
Also
you could format a long parameters list like that 
![image](https://github.com/ivandudardb/ddev/assets/109587214/8af613cb-ac81-4d3b-9aaa-fd3d877680c8)